src/idoswitchmenuitem.c

```diff
@@ -224,6 +224,14 @@ ido_source_menu_item_state_changed (IdoActionHelper *helper,
                            g_variant_get_boolean (state));
 }
 
+static void ido_switch_menu_item_activate(IdoSwitchMenuItem *pSelf, gpointer pUserData)
+{
+    IdoActionHelper *pHelper = pUserData;
```

Comment:
shouldn't we check some input here, e.g.
```
g_return_if_fail (IDO_IS_<type>(<input>));
```

Reply:
And why this passing around of helper, to pUserData, to pHelper?

Reply:
> And why this passing around of helper, to pUserData, to pHelper?

The **IdoActionHelper** is created locally in **ido_switch_menu_item_new_from_menu_model**, so we need to pass it as a user parameter to the callback. Because the callback receives a **gpointer**, we require a cast back to **IdoActionHelper**. The cast is implicit here.

Reply:
> shouldn't we check some input here, e.g.
> 
> ```
> g_return_if_fail (IDO_IS_<type>(<input>));
> ```

We could (as a sanity-check), but by the time we get here, **self** is guaranteed to be a fully initialised **IDO_SWITCH_MENU_ITEM**. If there were to be a failure, it would manifest elsewhere.

We lose nothing by adding stuff like this everywhere.

Reply:
> shouldn't we check some input here, e.g.
> 
> ```
> g_return_if_fail (IDO_IS_<type>(<input>));
> ```

Added via https://github.com/AyatanaIndicators/ayatana-ido/pull/15/commits/dc58df0d96e3ce3a43cded49317160564c5a9523